opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp

```diff
@@ -227,7 +227,19 @@ namespace Opm {
 
         using namespace std::placeholders;
         const auto handler = std::bind( handleKW, _1, schedule, props, n_xyz );
+
+#ifdef _MSC_VER
+        for (auto& x : section)
+        {
+            auto keywords = handler(x);
+            for (auto& keyword : keywords)
+            {
+                this->keywords.push_back(keyword);
+            }
+        }
+#else
         this->keywords = fun::concat( fun::map( handler, section ) );
```

Comment:
> it does, that's my point: the "plain old stupid" way with the for-loop would not require you to know anything about fun:: and boost::bind and it would not even be more lines ;)

It would most certainly be more lines, I don't know where you're getting that from. And it's not like you don't need to know the interfaces you use anyway, so that's not an argument.

> true, but it's only necessary if the new trick is sufficiently better than the old way, and that's not the case here, IMO. 

Your loss. You don't **have** to use it, but transforming-and-returning is a **common** operation, it will be used more in the future and a lot of operations today are essentially transform-and-return but are so convoluted in their manual approach that they don't appear that way anymore, and that removes the opportunity to generalise and simplify them. And fewer lines are typically easier to maintain than more lines.

Arguing that `map` is conceptually more difficult than declare container; iterate; append; is a *tough* sell, but feel free to give it a go.